tachyon/crypto/commitments/fri/two_adic_fri_pcs.h

```diff
@@ -0,0 +1,380 @@
+          for (size_t round = 0; round < num_rounds; ++round) {
+            const std::vector<
+                std::vector<std::tuple<ExtField, std::vector<ExtField>>>>&
+                claim = claims_by_round[round];
+            size_t vals_size = claim.size();
+            std::vector<math::Dimensions> batch_dims(vals_size);
+            size_t batch_max_height = 0;
+            for (size_t batch_idx = 0; batch_idx < vals_size; ++batch_idx) {
+              const Coset& mat_domain = domains_by_round[round][batch_idx];
+              size_t height = mat_domain.domain()->size() << fri_.log_blowup;
+              batch_max_height = std::max(batch_max_height, height);
+              batch_dims[batch_idx] = math::Dimensions{0, height};
+            }
+            size_t bits_reduced = log_global_max_height -
+                                  base::bits::StrictLog2(batch_max_height);
+            size_t reduced_index = index >> bits_reduced;
+            const std::vector<std::vector<F>>& opened_values =
+                input_proof[round].opened_values;
+
+            CHECK(mmcs_.VerifyOpeningProof(commits_by_round[round], batch_dims,
+                                           reduced_index, opened_values,
+                                           input_proof[round].opening_proof));
+
+            for (size_t batch_idx = 0; batch_idx < vals_size; ++batch_idx) {
+              const Coset& mat_domain = domains_by_round[round][batch_idx];
+              const std::vector<std::tuple<ExtField, std::vector<ExtField>>>&
+                  mat_points_and_values = claim[batch_idx];
+              size_t log_height =
+                  mat_domain.domain()->log_size_of_group() + fri_.log_blowup;
+              size_t bits_reduced = log_global_max_height - log_height;
+              size_t rev_reduced_index =
+                  base::bits::ReverseBitsLen(index >> bits_reduced, log_height);
+              F w;
+              CHECK(F::GetRootOfUnity(1 << log_height, &w));
+              F x = F::FromMontgomery(F::Config::kSubgroupGenerator) *
+                    w.Pow(rev_reduced_index);
+
+              if (reduced_openings.find(log_height) == reduced_openings.end()) {
+                reduced_openings.try_emplace(
+                    log_height,
+                    std::make_tuple(ExtField::One(), ExtField::Zero()));
+              }
+              for (size_t i = 0; i < mat_points_and_values.size(); ++i) {
+                const ExtField& z = std::get<0>(mat_points_and_values[i]);
+                const std::vector<ExtField>& ps_at_z =
+                    std::get<1>(mat_points_and_values[i]);
+                CHECK_EQ(ps_at_z.size(), opened_values[i].size());
+                for (size_t j = 0; j < ps_at_z.size(); ++j) {
+                  const ExtField& p_at_z = ps_at_z[j];
+                  ExtField quotient =
+                      unwrap((-p_at_z + ExtField(opened_values[batch_idx][j])) /
+                             (-z + ExtField(x)));
```

Comment:
I confirmed that the referenced code was written in the same way. Is there a special reason why the term with the minus operator must come first?

```suggestion
                      unwrap((ExtField(opened_values[batch_idx][j]) - p_at_z) /
                             (ExtField(x) - z));
```

Reply:
Oh I don't think there's a reason for this. I'll change it especially since it causes fewer calculations to occur.